### PR TITLE
Issue #5: Users can join existing shopping lists.

### LINKED ISSUE #5: 5. As a user, I want to join an existing shopping list so I can share a shopping list with another person.
Tokens can be shared with other users to allow them to co-manage a given list.

Joining an existing list consists of the following:

* User enters an existing token
* Save the token to `localStorage`
* Show the user the existing list

AC:

* When the user does not already have a token in `localStorage`, on the onboarding/home screen, a simple form is displayed that allows the user to enter a token
* Entering the token and hitting submit saves the token to `localStorage`, effectively giving them joint control of the list
* On submit, show an error if the token does not exist
* If they get an error message, allow them to try again or create a new list

### PR DESCRIPTION
**Pull request by hajarNasr**
## Type of Changes

|      | Type                       |
| --- | -------------------------- |
|  ✓    | :sparkles: New feature     |
|  ✓    |  🔨  Refactoring  |

## Acceptance Criteria
- [X] When the user does not already have a token in localStorage, on the onboarding/home screen, a simple form is displayed that allows the user to enter a token.
- [X] Entering the token and hitting submit saves the token to localStorage, effectively giving them joint control of the list.
- [X] On submit, show an error if the token does not exist.
- [X] If they get an error message, allow them to try again or create a new list.
## Related Issue
Closes #5

## Description
This PR does three things:

1- **It adds four new components**:
* `Header`: Just a simple header to the app, does nothing for now.
* `JoinAShoppingList`: A component that contains a form that allows users to enter shared tokens. A component that's responsible for making sure that only users with existing tokens get access to the shopping lists associated with these tokens.
* `MainIterface`: A component works like a homepage for unauthorized users. It contains a button that allows new users to generate tokens so that they can use the app. It also contains the `JoinAShoppingList` component. 
* `Form`: A component used as an attempt to reduce repetitive code.

2- **It adds a little bit of style to the app.**

3- **It removes**:
* The homeNavLink.
* The list route. 

## Demo
a demo for the app

## Testing Steps / QA Criteria
* From your terminal, pull down this branch with `git pull origin yk-hn-issue-5` and check that branch out with `git checkout yk-hn-issue-5` 

* Run `yarn start` to run the [local server](http://localhost:3000/).

* Try to join a shopping list with a fake token and then you should see this error msg: *Shopping list does not exist*.
* Try out this token: **dahlia term thug**, which is a token that has a shopping list in the database. By submitting that token you should have access to that shopping list.
If you still see that error msg, that means that you're out of luck and that someone must've cleared the database so that token no longer exists. If that happens, just let us know and we'll give you another one. We literally have a list of like 7000 words just for these token.

### CODE REVIEW

File: src/components/Navigation.js [hunk 1]
```diff
@@ -5,14 +5,11 @@ import '../css/components/Navigation.css';
...
   return (
     <div className="navlinks">
-      <NavLink activeClassName="selected" className="not-selected" to="/" exact>
-        Home
-      </NavLink>
       {hasToken && (
         <NavLink
           activeClassName="selected"
           className="not-selected"
-          to="/list"
```
Comment (deboragaleano) [inline src/components/Navigation.js:15]: I definitely agree with having only the `list` and the `add` links here! 👌

File: src/components/AddItem.js [hunk 2]
```diff
@@ -1,46 +1,34 @@
...
       howSoon: e.target['how-soon'].value,
     };
 
-    e.preventDefault();
-
-    shoppingLists
-      // fetch the token from localSorage
-      .where('token', '==', getToken())
-      .get()
+    getShoppingList(getToken())
```
Comment (deboragaleano) [inline src/components/AddItem.js:17]: nice! the `getShoppingList` function is now fetching the token 👍

File: src/components/Header.js [hunk 3]
```diff
@@ -0,0 +1,11 @@
+import React from 'react';
+
+const Header = () => {
```
Comment (deboragaleano) [inline src/components/Header.js:3]: I like this new header component 👏  
I'm starting to think what to say there for when we merge (maybe the app name?) 🤔

File: src/components/MainInterface.js [hunk 4]
```diff
@@ -0,0 +1,30 @@
+import React from 'react';
+import generateToken from '../lib/tokens';
+import { setToken } from '../lib/TokenService';
+import JoinAShoppingList from './JoinAShoppingList';
+
+const MainIterface = ({ setHasToken }) => {
```
Comment (deboragaleano) [inline src/components/MainInterface.js:6]: Oh I see, this is kind of replacing the `home` component, right?  I think it looks so much better now !

**Review by deboragaleano — COMMENTED** [review]
Everything looks great to me, especially on the browser!! And everything works as expected 👏🏽
Thanks so much for improving the style! 💟  😃

File: src/components/Header.js (hunk 3 above)
Comment (hajarNasr) [reply src/components/Header.js:3]: Yes! We need to come up with a cool name for the app. 😃

File: src/components/JoinAShoppingList.js [hunk 5]
```diff
@@ -0,0 +1,30 @@
+import React, { useState } from 'react';
+import { getShoppingList } from '../lib/shoppingListsCollection';
+import Form from './Form';
+
+const JoinAShoppingList = ({ onSharedToken }) => {
+  const [error, setError] = useState(false);
+
+  const handleSubmit = (e, inputValue) => {
+    getShoppingList(inputValue).then((data) => {
```
Comment (saraheisa) [inline src/components/JoinAShoppingList.js:9]: Good approach to check if there's a token or not 👍

**Review by saraheisa — APPROVED** [review]
Everything is running smoothly and as expected 
can't think of any comments Well Done 🙌
& I like the gif in the PR description Super Cool!

File: src/components/MainInterface.js [hunk 6]
```diff
@@ -6,13 +6,15 @@ import JoinAShoppingList from './JoinAShoppingList';
 
 const MainIterface = ({ setHasToken }) => {
   let history = useHistory();
-  const handleSettingToken = (token) => {
-    if (!token) {
+
+  const handleSettingToken = (sharedToken) => {
```
Comment (llunaplanet) [inline src/components/MainInterface.js:7]: This change from **token** to **sharedToken** make things clearer 👍

**Review by llunaplanet — APPROVED** [review]
All the acceptance criteria has been meet, everything working as expected 🚀 , the mini gif demo is super cool.
I see some interesting refactors on naming to make things clearer, good work 🙌!

File: src/App.js [hunk 7]
```diff
@@ -3,19 +3,20 @@ import { BrowserRouter, Switch, Route } from 'react-router-dom';
 import List from './components/List';
```
Comment (llunaplanet) [inline src/App.js:3]: I see some warnings that this **List** is defined but not used

Comment (hajarNasr) [reply src/App.js:3]: I think we forgot to remove it when we removed the list route in that file. We'll remove it when we merge. :) 
Thank you so much for the review. ❤️

**Pull request merged**